torch/distributed/pipelining/stage.py

```diff
@@ -1257,25 +1268,25 @@ def __init__(
         group: Optional[dist.ProcessGroup] = None,
         dw_builder: Optional[Callable[[], Callable[..., None]]] = None,
     ):
+        assert submodule.device == torch.device(
```

Comment:
this assertion breaks tests. i guess itll break some existing usages.  Is it a good thing to do to prevent bad practices in the future, or should i relax this and make it work if model+inputs are on cuda?

Reply:
I think we should relax this. "meta" device is still a high-end thing for most users.
And, is `nn.Module.device` a real thing?

Reply:
hmm, its not always possible to assert this- maybe there is another way to reliably check the device of the module?

` AttributeError: 'FSDPSequential' object has no attribute 'device'`

Reply:
yea, maybe i should just drop the checks and perform the shape inference blindly, but then do a .to(meta) on the outputs.  that should make things on par for what howard mentioned, and then we can delete it once we add lazy inference

Reply:
`FSDPSequential` is a superset of 'nn.Sequential'. It means vanilla `nn.Sequential` does not have `device`. I guess what we really want to check whether `model.parameters()` are on meta? that should cover all cases

```
>>> import torch
>>> model = torch.nn.Sequential(torch.nn.Linear(2,2))
>>> model.device
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/users/weif/pytorch/torch/nn/modules/module.py", line 1931, in __getattr__
    raise AttributeError(
AttributeError: 'Sequential' object has no attribute 'device'
```